source/extensions/filters/http/gzip/gzip_filter.cc

```diff
@@ -72,14 +76,46 @@ uint64_t GzipFilterConfig::windowBitsUint(Protobuf::uint32 window_bits) {
 
 const envoy::extensions::filters::http::compressor::v3::Compressor
 GzipFilterConfig::compressorConfig(const envoy::extensions::filters::http::gzip::v3::Gzip& gzip) {
+  envoy::extensions::filters::http::compressor::v3::Compressor compressor = {};
+
+  // The deprecated gzip filter is using a locally defined Compressor field to avoid setting the
+  // compressor_library field which is mandatory in v3. Here we convert gzip's Compressor to
+  // what is accepted by the generic compressor.
```

Comment:
It's up to you, but you might consider doing a YAML/JSON round trip which might be easier to read and understand (see the protobuf message utlities in protobuf/utility.h. We do this in other places. As this is not perf critical it's probably fine.

Reply:
Thank you! I didn't know about this utility. With the round trip the code is easier to read indeed.